python/cudf/cudf/core/reshape.py

```diff
@@ -681,3 +683,212 @@ def merge_sorted(
     )
     result._copy_categories(objs[0])
     return result
+
+
+def _pivot(df, index, columns):
+    """
+    Reorganize the values of the DataFrame according to the given
+    index and columns.
+
+    Parameters
+    ----------
+    df : DataFrame
+    index : cudf.core.index.Index
+        Index labels of the result
+    columns : cudf.core.index.Index
+        Column labels of the result
+    """
+    columns_labels, columns_idx = columns._encode()
+    index_labels, index_idx = index._encode()
+    column_labels = columns_labels.to_pandas().to_flat_index()
+
+    # the result of pivot always has a multicolumn
+    result = cudf.core.column_accessor.ColumnAccessor(
+        multiindex=True, level_names=(None,) + columns._data.names
+    )
+
+    def as_tuple(x):
+        return x if isinstance(x, tuple) else (x,)
```

Comment:
I thought about it, but it could be abused/misused. For example, if `x` is already something like a tuple it returns an object that could lead to bugs. That's why I decided to just keep it local

Reply:
Whereas within the context of this function, we're assured that the input is always either a `tuple` or a column name like a string.